lib/note-content-editor.jsx

```diff
@@ -171,6 +220,29 @@ export default class NoteContentEditor extends React.Component {
 		)
 	}
 
+	handleReturn = () => {
+		// matches lines that start with `- ` or `* `
+		// preceded by 0 or more tab characters
+		const listItemRe = /^\t*[-*]\s/;
```

Comment:
I haven't tried out the PR, but no indent should be required imo. Meaning:

1. type dash character + some text (With space or no space, doesn't matter. Tab creates indented list)
2. press enter
3. new line is created with dash

This is how the Mac app works, and I think it's fine. iOS app works the same way.